docs/TuxMutableClass.md

```diff
@@ -0,0 +1,201 @@
+
+A specific example is if you have tens or hundreds of message components rendering on the screen. The mutable class would be beneficial at that point because most of those message components can update, but will not necessarily update with great frequency. Therefore having components that will check to see if they should re-render in the virtual DOM can result in a performance boost with the more messages that are on the screen. If someone edits the text of a message, instead of re-rendering all of the other messages in the virtual DOM with the mutable class only the one message that has updated will do so.
+
+```javascript
+    // GOOD
+
+
+
+    // BAD
+```
+
+Taking all that we have learned so far let's take a look at a more realistic example.
+
+## <a id="TuxMutableClass-Complete-Example"></a>TuxMutableClass Complete Example [#](#TuxMutableClass-Complete-Example)
+
+```javascript
+    var React = require('tux/React');
+    var MessageForm = require('./MessageForm.jsx'); // Message form that we will pass a message to edit to
+
+    var Message = React.createMutableClass({
+      mutableTraits: {
+        props: [['message', 'text']],
+        state: 'editing'
+      },
+
+      getInitialState: function () {
+        return {
+          editing: false
+        };
+      },
+
+      componentWillReceiveProps: function () {
+        this.closeEditForm();
+      },
+
+      closeEditForm: function() {
+        this.setState({ editing: false });
+      },
+
+      deleteMessage: function(e) {
+        e.preventDefault();
+        this.nearestOwnerProps.deleteMessage(this.props.message.id);
+      },
+
+      updateMessage: function (message) {
+        this.nearestOwnerProps.updateMessage(message, this.props.message.id);
+        this.closeEditForm();
+      },
+
+      edit: function(e) {
+        e.preventDefault();
+        if (this.isMounted()) {
+          this.setState({ editing: !this.state.editing });
+        }
+      },
+
+      render: function () {
+        var editForm;
+        var message = this.props.message;
+
+        if (this.state.editing) {
+          editForm = <MessageForm message={message} editing={this.state.editing} roomId={this.props.roomId} addOrEdit='Edit' />
+        }
+
+        return (
+          <li>
+            {message.username} - {message.text} <br />
+            {editForm}
+            <button onClick={this.deleteMessage}>Delete</button><button onClick={this.edit}>Edit</button>
+          </li>
+        );
+      }
+    });
+```
+
+In this example the Message component takes advantage of the mutable class by registering the `editing` property on state and a path on props that will contain `message` + `text` via the use of the `mutableTraits` property. This will the paths necessary to watch these paths the first time a component of the Message component class is mounted.
```

Comment:
Change to ...`by registering `this.state.editing` and `this.props.message.text` as the mutable traits of this component using the `mutableTraits` property. Whenever this component will be triggered to update it will check to see if `this.state.editing` and/or `this.props.message.text` has updated.  If neither has then `shouldComponentUpdate` will return false and the component will not need to re-render in the virtual DOM.